File: Data Structures & Algorithms/serialize-and-deserialize-binary-tree/submission-0.java
public class Codec {

    // Encodes a tree to a single string (Preorder DFS)
    public String serialize(TreeNode root) {
        StringBuilder sb = new StringBuilder();
        buildString(root, sb);
        return sb.toString();
    }

    private void buildString(TreeNode node, StringBuilder sb) {
        if (node == null) {
            sb.append("null,");
            return;
        }
        sb.append(node.val).append(",");
        buildString(node.left, sb);
        buildString(node.right, sb);
    }

    // Decodes your encoded data to tree
    public TreeNode deserialize(String data) {
        String[] arr = data.split(",");
        java.util.Queue<String> queue = new java.util.LinkedList<>();
        for (String s : arr) {
            queue.offer(s);
        }
        return buildTree(queue);
    }

    private TreeNode buildTree(java.util.Queue<String> queue) {
        String val = queue.poll();
        if (val.equals("null")) return null;

        TreeNode node = new TreeNode(Integer.parseInt(val));
        node.left = buildTree(queue);
        node.right = buildTree(queue);
        return node;
    }
}

File: Data Structures & Algorithms/delete-leaves-with-a-given-value/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @param {number} target
     * @return {TreeNode}
     */
    removeLeafNodes(root, target) {
        if (!root) return null;

        root.left = this.removeLeafNodes(root.left, target);
        root.right = this.removeLeafNodes(root.right, target);

        if (
            !root.left &&
            !root.right &&
            root.val === target
        ) {
            return null;
        }

        return root;
    }
}